conda/petsc/build.sh

```diff
@@ -24,7 +24,7 @@ if [[ $(uname) == Darwin ]]; then
         CTUNING="-march=core2 -mtune=haswell"
         FTUNING="-I$PREFIX/include"
     fi
-    LDFLAGS="${LDFLAGS:-} -Wl,-headerpad_max_install_names"
+    LDFLAGS="${LDFLAGS:-} -Wl,-headerpad-max-install-names"
```

Comment:
@milljm @dschwen Looking at the M1 PETSc configure error you posted reminded me of this needed change on my Intel Mac, which threw a similar error. Not sure what caused it yet (I assumed originally we were just using an old form of the flag), but I wanted to make note.